assets/pages/regular-contributions/helpers/ajax.js

```diff
@@ -94,83 +96,123 @@ function requestData(
     supportAbTests,
   };
 
-  if (state.user.stateField) {
-    regularContribFields.state = state.user.stateField;
+  if (user.stateField) {
+    regularContribFields.state = user.stateField;
   }
 
   return {
     method: 'POST',
-    headers: { 'Content-Type': 'application/json', 'Csrf-Token': state.csrf.token || '' },
+    headers: { 'Content-Type': 'application/json', 'Csrf-Token': csrf.token || '' },
     credentials: 'same-origin',
     body: JSON.stringify(regularContribFields),
   };
 }
 
-function statusPoll(dispatch: Function, getState: Function) {
-  const state = getState();
+let trackingURI = null;
+let pollCount = 0;
```

Comment:
Should this state be in redux?

Reply:
I would love to put it in the redux store. Unfortunately, since we removed the redux state from `StripePopUpButton`, I haven't come up with a solution to have the getState function inside `ajax.js`. Before we used to inject that function by `mapDispatchToProps` but with no connect function that approach is unfeasible.